Assets/Plugins/Source/Core/Utility/EnumUtility.cs

```diff
@@ -197,5 +197,37 @@ private static TEnum Combine(TEnum current, TEnum toAdd)
 
             throw new ArgumentException($"Unsupported enum underlying type: \"{underlyingType.FullName}.\"");
         }
+
+        /// <summary>
+        /// Enum types that have the attribute [Flags], and have a compatible
+        /// underlying type can be combined using bitwise operations. This
+        /// utility function will take that combined value and decompose it into
+        /// the individual Enum values represented by the value.
+        ///
+        /// This is mostly useful for debugging and for displaying the value in
+        /// user interfaces, such as instances where the user can change the
+        /// values.
+        /// </summary>
+        /// <param name="bitFlag">
+        /// The possible result of combining enum values using bitwise
+        /// operations.
+        /// </param>
+        /// <returns>
+        /// The discrete values of which this combined flag is composed of.
+        /// </returns>
+        public static IEnumerable<TEnum> Decompose(TEnum bitFlag)
```

Comment:
This looks like a non config specific thing. Maybe it can be in a "Core" adjacent class

Reply:
It is a bit of a gray area. My thought process was that the `RuntimeConfig` object would not be able to display a user interface for an enum value that was the result of a bitwise OR operation on multiple values. This code was included to make it clear that aspect of the change was considered. You are correct that it's not strictly speaking config related, even though it will exclusively be used within the context of config operations.

It's a small enough case that I'm comfortable including it, since it doesn't _change_ the functionality of anything, as the function is not called. But I could go either way on it. Lemme know your thoughts and I'll take action.

Reply:
I think I prefer names of the style `ToEnumerable` or something similar than 'Decompose'. I think it's because `Decompose` is too similar to `Dispose`.